### ISSUE TITLE
Has XLM been experimented with mixed-XNLI (parallel inputs from different languages)?

### ISSUE THREAD

**Issue by Akella17** [facebookresearch/XLM#90]
I have noticed that the XLM model, fine-tuned on English corpora, provides a SOTA for zero-shot cross-lingual classification on XNLI. However, the entire input text stream is from the same language (if I have understood the paper correctly).

Has there been any NLI experiment with the premise and hypothesis coming from different languages? Since none of the LM pre-training tasks uses a stream of mixed language as inputs (even TLM pre-training uses synonymous sentences), how do we know that the model can handle a continuous stream of mixed-language data (assuming language ID tags are provided for the input tokens) at test time?

**Comment by zyuuyuu** [comment]
Echo this question.

I am trying to train a multilingual classifier and training on my English data set does not have a good zero-shot prediction on the German data set I have. 

So I tried to feed in both the German and English dataset into training, while still brewing, the accuracy at the first sight is very low. 

Just wondering if you guys did similar experiment...

**Comment by aconneau** [comment]
We did not do experiments where you have a French (say) premise and a Chinese hypothesis no. Although this is indeed of the possibilities of the XNLI corpus. I would expect that the MLM+TLM model would do a good job there, as it has been trained to handle concatenations of sentences coming from different languages. However it has only seen parallel sentences. If you concatenante more than two sentences coming from various languages, I have no idea. It would probably not work so well as the model has never been trained to handle this but it could still figure out a way to handle that task nicely if you have a sufficiently large training set for your downstream task. Or you can pretrain your XLM with a TLM that handles more that pairs of sentences (by using n-way parallel corpora like Europarl or MultiUN).
